tidy.Rmd

```diff
@@ -124,54 +124,120 @@ Typically a dataset will only suffer from one of these problems; it'll only suff
 
 ```{r}
 table4a
 ```
 
+And you want to create the following visualisation where each line represents a `country`, `year` is on the x-axis, `cases` are on the y-axis, and you automatically get the legend that indicates which line represents which country.
+
+```{r tidy-pivot-longer-plot-lines, fig.width = 5, echo = FALSE}
+table4a %>%
+  pivot_longer(
+    cols = c(`1999`, `2000`),
+    names_to = "year",
+    names_transform = list(year = as.integer),
+    values_to = "cases",
+  ) %>%
+  ggplot(aes(x = year, y = cases)) +
+  geom_line(aes(group = country), colour = "grey50") + 
+  geom_point(aes(colour = country, shape = country)) +
+  scale_x_continuous(breaks = c(1999, 2000))
+```
+
+It's most straight-forward to do this starting with a data frame where `country`, `year`, and `cases` are the columns and each row represents a record from a country for a particular year.
+
+```{r echo = FALSE}
+table4a %>%
+  pivot_longer(
+    cols = c(`1999`, `2000`),
+    names_to = "year",
+    names_transform = list(year = as.integer),
+    values_to = "cases"
+  )
+```
+
+However in `table4a` the column names `1999` and `2000` represent values of the `year` variable, the values in the `1999` and `2000` columns represent values of the `cases` variable, and each row represents two observations, not one.
+
 To tidy a dataset like this, we need to __pivot__ the offending columns into a new pair of variables. To describe that operation we need three parameters:
 
 * The set of columns whose names are values, not variables. In this example, 
   those are the columns `1999` and `2000`.
 
-* The name of the variable to move the column names to. Here it is `year`.
+* The name of the variable to move the column names to: `year`.
 
-* The name of the variable to move the column values to. Here it's `cases`.
+* The name of the variable to move the column values to: `cases`.
   
 Together those parameters generate the call to `pivot_longer()`:
 
 ```{r}
 table4a %>% 
-  pivot_longer(c(`1999`, `2000`), names_to = "year", values_to = "cases")
+  pivot_longer(
+    cols = c(`1999`, `2000`),
+    names_to = "year", 
+    values_to = "cases"
+    )
 ```
 
-The columns to pivot are specified with `dplyr::select()` style notation. Here there are only two columns, so we list them individually. Note that "1999" and "2000" are non-syntactic names (because they don't start with a letter) so we have to surround them in backticks. To refresh your memory of the other ways to select columns, see [select](#select).
+The columns to pivot are specified with `dplyr::select()` style notation in the `cols` argument. Here there are only two columns, so we list them individually. Note that `1999` and `2000` are non-syntactic names (because they don't start with a letter) so we have to surround them in backticks. To refresh your memory of the other ways to select columns, see [select](#select).
```

Comment:
Should probably change cross-links to `Section @\ref(#select)`.

Reply:
Fixed throughout the chapter